p2p/agent.go

```diff
@@ -92,8 +93,15 @@ func (p *Agent) Start(ctx context.Context) error {
 	}
 
 	if err := host.AddBroadcastPubSub(broadcastTopic, func(data []byte) (err error) {
+		// Blocking handling the broadcast message until the agent is started
+		<-ready
```

Comment:
where is the wring to "ready"?

Reply:
It is used to block the call back function to move forward. The goroutine will be blocked here if the callback is triggered, until L178 `ready` is closed